src/docfx/build/metadata/ContributionInfo.cs

```diff
@@ -52,57 +56,80 @@ public static ContributionInfo Load(Docset docset, string gitToken)
             Debug.Assert(document != null);
 
             var errors = new List<Error>();
-            UserProfile authorInfo = null;
-            if (!string.IsNullOrEmpty(author))
+
+            _commitsByFile.TryGetValue(document.FilePath, out var commits);
+            var authorInfo = await GetAuthor(author, commits);
+            var contributors = GetContributors(commits);
+            var updatedDateTime = GetUpdatedAt(commits);
+
+            return (errors, ToGitUserInfo(authorInfo), contributors.Select(ToGitUserInfo).ToArray(), updatedDateTime);
+
+            async Task<UserProfile> GetAuthor(string authorName, List<GitCommit> fileCommits)
```

Comment:
>, List<GitCommit> fileCommits [](start = 63, length = 29)

nit: if this is a nested function, you can just use commits from parent scope.

Reply:
make the input and output clear.

---
In reply to: [210816527](https://github.com/dotnet/docfx/pull/3224#discussion_r210816527) [](ancestors = 210816527)